From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Wed, 4 Jan 2012 15:05:10 -0800
Subject: driver core: remove __must_check from device_create_file

commit b9d4e714a86a4e88c2f530c76597f7025e5851d6 upstream.

With the conversion of the sysdev to a real struct device, more drivers
are calling device_create_file, and some of them don't check the return
value, which isn't wise.

But as they happen to be in parts of the kernel where a warning is
considered an error (i.e. powerpc), this breaks the build.  So for now,
remove the marking on the function, which fixes the build problems.

Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 include/linux/device.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-3.2.46/include/linux/device.h
===================================================================
--- linux-3.2.46.orig/include/linux/device.h	2013-07-26 19:22:51.000000000 +0000
+++ linux-3.2.46/include/linux/device.h	2013-07-26 19:23:37.000000000 +0000
@@ -510,8 +510,8 @@
 	struct dev_ext_attribute dev_attr_##_name = \
 		{ __ATTR(_name, _mode, device_show_ulong, device_store_ulong), &(_var) }
 
-extern int __must_check device_create_file(struct device *device,
-					const struct device_attribute *entry);
+extern int device_create_file(struct device *device,
+			      const struct device_attribute *entry);
 extern void device_remove_file(struct device *dev,
 			       const struct device_attribute *attr);
 extern int __must_check device_create_bin_file(struct device *dev,
